pluginLoader.js

```diff
@@ -33,7 +33,12 @@ define(function(require, exports, module){
             var _lint = plugin.lint;
 
             function lint(text, settings) {
-                return Promise.resolve(_lint(text, settings));
+                var results = _lint(text, settings,pluginsMeta);
+                if(results.promise !== undefined && results.promise !== null){
```

Comment:
@jwestbrook apologies for the terribly slow response.

Promise.resolve will gracefully convert input objects to promises... And input promises are basically a noop.  That's a good way to uniformly handle promises and non promises objects consistently.

``` javascript
return Promise.resolve(_lint(text, settings, pluginsMeta));
```

Should be sufficient.